modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/provider/test/APIApplicationProviderTest.java

```diff
@@ -162,6 +165,46 @@ public void test() throws Exception {
 
 		Assert.assertEquals(
 			"name ne 'testName'", filter.getODataFilterString());
+
+		String baseURL = RandomTestUtil.randomString();
+
+		HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"applicationStatus", "published"
+			).put(
+				"baseURL", baseURL
+			).put(
+				"title", "test"
+			).toString(),
+			"headless-builder/applications/" +
+				apiApplicationJSONObject.getLong("id"),
+			Http.Method.PUT);
+
+		apiApplication = _apiApplicationProvider.fetchAPIApplication(
+			baseURL, TestPropsValues.getCompanyId());
+
+		Assert.assertEquals(baseURL, apiApplication.getBaseURL());
+		Assert.assertEquals("test", apiApplication.getTitle());
+
+		HTTPTestUtil.invokeToJSONObject(
```

Comment:
I'm just taking a look at `HeadlessBuilderResourceTest` and the` testGet()` there's a method `_publishAPIApplication(_API_APPLICATION_ERC_1);` and later on the assert of the code of this PATCH operation. Then later on it is being unpublished again via ERC (again by PATCH operation).  So I wonder if we should add these tests there as well 😅